src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -18,17 +18,27 @@ namespace Microsoft.IdentityModel.Protocols
     /// </summary>
     /// <typeparam name="T">The type of <see cref="IDocumentRetriever"/>.</typeparam>
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable")]
-    public class ConfigurationManager<T> : BaseConfigurationManager, IConfigurationManager<T> where T : class
+    public partial class ConfigurationManager<T> : BaseConfigurationManager, IConfigurationManager<T> where T : class
     {
-        // To prevent tearing, this needs to be only updated through AtomicUpdateSyncAfter.
-        // Reads should be done through the property SyncAfter.
+#pragma warning disable IDE0044 // Add readonly modifier
+#pragma warning disable CS0649 // Unused, it gets used in tests.
+        internal Action _onBackgroundTaskFinish;
```

Comment:
I suggest that internals should have the same casing as public members, since we kind of treat internals and publics similar related to breaking changes. Also seeing an underscore make it easier to think about code since that variable is private and not used anywhere else.

We aren't really consistent with this naming throughout our SDKs though (using both underscore and Pascal casing). We could update our rules to issue a warning, but that will probably require suppressions, since we wouldn't want to rename internal members. Thoughts?

```suggestion
        internal Action OnBackgroundTaskFinish;
```

Reply:
Ha, in this case, not having the underscore raises an error, would rather not suppress it here.